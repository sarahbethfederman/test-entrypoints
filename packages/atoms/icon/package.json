{
  "name": "@auscred/icon",
  "version": "0.0.1",
  "source": "src/Icon.tsx",
  "main": "dist/Icon.js",
  "types": "dist/Icon.d.ts",
  "directories": {
    "lib": "lib"
  },
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "devDependencies": {
    "jest": "^23.1.0",
    "@lui/story-utils": "^1.0.0",
    "@storybook/addon-knobs": "^3.4.8",
    "@types/storybook__addon-knobs": "^3.3.1"
  },
  "scripts": {
    "build": "node ./src/build.js && tsc -p ./tsconfig.json"
  },
  "dependencies": {
    "svg-sprite": "^1.4.0"
  }
}
