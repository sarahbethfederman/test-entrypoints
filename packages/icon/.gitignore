src/icons-compiled/*
src/index.ts