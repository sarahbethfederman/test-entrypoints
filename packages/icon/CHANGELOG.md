## [9.6.2] - 2020-09-02 - [SUB-609][https://creditandfinance.atlassian.net/browse/sub-609]

## 10.0.0-beta.0

### Minor Changes

- preconstruct infra update

### Patch Changes

- Updated dependencies [undefined]
  - @lendi-ui/color@6.0.0-beta.0
  - @lendi-ui/theme@10.0.0-beta.0

### Added

- Added the domain bot Icon && Updated the autobot Icon name to LendiBotIcon

## [9.5.2] - 2020-08-27 - [SUB-601][https://creditandfinance.atlassian.net/browse/sub-601]

### Updated

- Updated Domain Home Loans for Team View

## [9.5.1] - 2020-08-18 - [SUB-461][https://creditandfinance.atlassian.net/browse/sub-461]

### Added

- Added Domain Home Loans logos

## [9.5.0] - 2020-04-28 - [SUB-242][https://creditandfinance.atlassian.net/browse/sub-242]

### Added

- Added autobot icon

## [9.4.1] - 2020-04-28 - [SUB-hotfix-unreadMessageClass]

### Added

- Updated the fillPath class on the unread message icon.

## [9.4.0] - 2020-04-28 - [SUB-hotfix-redDot]

### Added

- Added red dot icon.

## [9.3.0] - 2020-03-18 - [HUB-hotfix-domainIcon](https://creditandfinance.atlassian.net/browse/FUNNEL-1081)

### Added

- Added domain logomark icon.
- Added domain teamview logo icon.

## [9.2.0] - 2020-02-19 - [HUB-893](https://creditandfinance.atlassian.net/browse/HUB-893)

### Added

- Added unread icon.

## [9.1.1] - 2019-11-22

### Changed

- Fixed the colours in the new PropertyReport icon.

## [9.1.0] - 2019-11-22 - [HUB-664](https://creditandfinance.atlassian.net/browse/HUB-664)

### Added

- Added in relation to: HUB-664, HUB-665 and HUB-695
- Adds: ArrowRightCircle, BullseyeArrow, LightbulbOn, HomeMapMarker, HomeAccount, PropertyReport, CalculatorVariant, IcAssignment and CalendarMonth icons.

## [9.0.2] - 2019-09-27

- Change color to an optional property.

## [9.0.1] - 2019-09-26 [Release Notes](https://creditandfinance.atlassian.net/wiki/spaces/HUB/pages/803930391/Upcoming+Major+Changes)

- Bumped lendi-ui dependencies to latest versions

## [9.0.0] - 2019-9-23 [Release notes](https://creditandfinance.atlassian.net/wiki/spaces/HUB/pages/803930391/Upcoming+Major+Changes)

### Changed

- Uses Theme v7
- Update typescript to version 3.5.2 and react to version 16.8.4 [HUB-151](https://creditandfinance.atlassian.net/browse/HUB-151)

## [8.2.0] - 2019-07-16 - [HUB-330](https://creditandfinance.atlassian.net/browse/HUB-330)

### Changed

- Change icon default color to 'currentColor'

## [8.1.0] - 2019-06-07 - [HUB-302](https://creditandfinance.atlassian.net/browse/HUB-302)

### Changed

- Added the vertical-split icon

## [8.0.1] - 2019-06-07 - [HUB-264](https://creditandfinance.atlassian.net/browse/HUB-264)

### Changed

- Fix the icon alignment issue.

## [8.0.0] - 2019-05-29 - [HUB-144](https://creditandfinance.atlassian.net/browse/HUB-144)

### Deleted

- FileDownload, SuccessTwo, Warn, Error, Mobile (duplicate icons)
- All fancy icons (moved to the new fancy-icon package)

## Changed

- Renamed icons to match material.ui.

## [7.2.0] - 2019-05-17 - [HUB-174](https://creditandfinance.atlassian.net/browse/HUB-174)

### Changed

- Module format reverted back to CJS.

### Deleted

- Dropped "module" property from package.json.

## [7.1.2] - 2019-05-09 - [HUB-92](https://creditandfinance.atlassian.net/browse/HUB-92)

### Changed

- Updated CHANGELOG to match required parsing format.

## [7.1.1] - 2019-05-08 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Changed

- Change module main entry point to esm in package.json

## [7.1.0] - 2019-05-02 - [HUB-126](https://creditandfinance.atlassian.net/browse/HUB-126)

### Added

- Added the following new icons: file-upload & FileDownload as an alias to GetApp.

## [7.0.1] - 2019-05-02 - [HUB-132](https://creditandfinance.atlassian.net/browse/HUB-132)

### Changed

- Fix MS's Edge Icon rendering issue

## [7.0.0] - 2019-04-30 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Deleted

- Remove the CJS bundles.

## [6.0.0] - 2019-04-16 - [HUB-49](https://creditandfinance.atlassian.net/browse/HUB-49)

### Changed

- Remove Left and Right Icons.

## [5.3.1] - 2019-04-08 - [HUB-71](https://creditandfinance.atlassian.net/browse/HUB-71)

### Changed

- Added styling to the icon container to align text and icon baseline when using inline icons.

## [5.3.0] - 2019-04-01 - [HUB-53](https://creditandfinance.atlassian.net/browse/HUB-53)

### Added

- Added the following new icons: attach-file, assignment-returned, autorenew, assessment, save, credit-card, assignment-return, people, more-horiz, done-all, assignment-turned-in, error2, search, date-range, assignment, open-with, dashboard, timeline, gavel, trending-flat, report-problem, trending-down, delete, fullscreen, delete-outline, remove-circle, sentiment-satisfied-alt, launch, edit, add-alert, trending-up, help, history, schedule, room, notification-important, get-app, share, highlight-off, more-vert, favorite-border, hourglass-empty, info2, monetization-on, thumb-up, help-outline, error-outline, add, check2, warning, build, computer, data-usage, alarm, description, stay-current-portrait, power-settings-new, account-balance

## [5.2.0] - 2019-03-15 - [BRAVO-576](https://creditandfinance.atlassian.net/browse/BRAVO-576)

### Added

- Added un-circled warn, info and success icons `info-two success-two warn-two`

## [5.1.0] - 2019-03-06 - [BRAVO-572](https://creditandfinance.atlassian.net/browse/BRAVO-572)

### Added

- Added Maintenance icon and PageNotFound icon.

## [5.0.1] - 2019-03-05 - [BRAVO-387](https://creditandfinance.atlassian.net/browse/BRAVO-387)

### Added

- Added Domain logo.

## [5.0.0]

### Changed

- Breaking change version update

## [4.4.0]

### Changed

- Bump version for left and right icons
