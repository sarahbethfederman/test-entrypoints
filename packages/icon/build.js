const SVGSpriter = require('svg-sprite');
const fs = require('fs');
const path = require('path');
const mkdirp = require('mkdirp');

const directory = `${__dirname}/src/icons`;

const spritify = new SVGSpriter({
  dest: 'dist',
  mode: {
    symbol: true,
    inline: true,
  },
});

fs.readdir(directory, (err, files) => {
  if (err) {
    console.error(err);
    return;
  }

  files.forEach((file) => {
    if (file.endsWith('.svg')) {
      spritify.add(
        path.resolve(`${directory}/${file}`),
        file,
        fs.readFileSync(`${directory}/${file}`, { encoding: 'utf-8' })
      );
    }
  });

  spritify.compile(function(error, result, data) {
    for (var mode in result) {
      for (var resource in result[mode]) {
        fs.writeFileSync(`../../website/static/sprite.symbol.svg`, result[mode][resource].contents);
      }
    }
  });
});
