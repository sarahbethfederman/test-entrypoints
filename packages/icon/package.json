{
  "name": "@lendi-ui/icon",
  "version": "9.3.0",
  "main": "dist/cjs/index.js",
  "source": "src/index.tsx",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^8.0.0",
    "@types/react": "^16.8",
    "@types/styled-components": "^4.1.18",
    "react": "^16.8",
    "styled-components": "^4"
  },
  "devDependencies": {
    "@svgr/core": "^2.2.0",
    "@lendi-ui/typography": "^4.0.0",
    "@lendi-ui/spacing": "^6.0.0"
  },
  "dependencies": {
    "@lendi-ui/color": "^5.0.0"
  },
  "scripts": {
    "build:svg": "node ./build/index.js",
    "build:cjs": "tsc --project tsconfig.cjs.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:svg && yarn run build:types && yarn run build:cjs"
  }
}
