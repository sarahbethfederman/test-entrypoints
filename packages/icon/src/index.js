import React from 'react';

const Icon = () => <span>Icon</span>;

export default Icon;
