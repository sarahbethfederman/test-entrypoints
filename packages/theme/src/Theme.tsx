import * as React from 'react';
import styled, { ThemeProvider, createGlobalStyle } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
const GlobalStyle = createGlobalStyle`
  html, body {
    margin: 0;
    padding: 0;
  }

  input::-ms-clear {
    display: none;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors['shade.700']};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>
        {children}
        <GlobalStyle />
      </Root>
    </ThemeProvider>
  );
}
