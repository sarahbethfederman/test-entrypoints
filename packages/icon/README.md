# @lendi-ui/icon

## Adding a new icon

Add the SVG into src/icons. Be thoughtful with naming, as this will determine the name of the component. For example, the file `computer-bank` will create a SVG component called `computerBank` (note that the kebab-case filename will become a camelCase component).

To add a new alias, navigate to the ./build/alias.json file and add the key as the original file name (in pascal case) and give it an array with all the alias' to add.

When this package is built, an equivalent react component will be created in src/icons-compiled.

## Installation

```
yarn add @lendi-ui/icon
```

## Usage

```jsx
import { Lock } from '@lendi-ui/icon';

<Lock color="secondary.500" />;
```

## Fill colours not working?

In this case, you will need to add a `fillPath` class to the element in the SVG that needs to be recoloured, this can be determined by inspecting the svg in the browser and changing the fill colour until you find the correct element.

## Properties

```
<PropTable>
  <PropTable.Entry required name="color" type={`string`} description="The Icon color"/>
</PropTable>
```
