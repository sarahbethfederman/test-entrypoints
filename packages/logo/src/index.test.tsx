import * as React from 'react';
import { mount } from 'enzyme';
import Logo from '.';
import Theme from '@lendi-ui/theme';
import { LendiLogo } from '@lendi-ui/icon';

/**
 * Test case Logo
 */
describe('Logo', () => {
  let Wrapper;
  /**
   * Test Logo wihout color, width and height
   */
  describe('with no color, width, height and className', () => {
    beforeEach(() => {
      Wrapper = mount(
        <Theme>
          <Logo />
        </Theme>
      );
    });
    it('should render "shade.1000" styles when no color is given', () => {
      expect(Wrapper).toMatchSnapshot();
    });
  });

  /**
   * Test Logo with color, width and height
   */
  describe('with provided color, width, height and className', () => {
    beforeEach(() => {
      Wrapper = mount(
        <Theme>
          <Logo variant="light" width="240px" height="120px" className="some-class" />
        </Theme>
      );
    });
    it('should render "shade.800" styles when color is given', () => {
      expect(Wrapper.find(LendiLogo).props().color).toEqual('shade.0');
    });

    it('should render width="240px" when width is given', () => {
      expect(Wrapper.find(Logo).props().width).toEqual('240px');
    });

    it('should render height="120px" when height is given', () => {
      expect(Wrapper.find(Logo).props().height).toEqual('120px');
    });

    it('should render className="some-class" when className is given', () => {
      expect(Wrapper.find(Logo).props().className).toEqual('some-class');
    });

    it('should render the icon with class "some-class" when className is given', () => {
      expect(Wrapper.find(LendiLogo).hasClass('some-class')).toEqual(true);
    });
  });
});
