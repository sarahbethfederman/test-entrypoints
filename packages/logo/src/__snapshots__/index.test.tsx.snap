// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo with no color, width, height and className should render "shade.1000" styles when no color is given 1`] = `
.c0 {
  width: 160px;
  height: 100px;
  vertical-align: middle;
  display: inline-block;
}

.c0 > svg {
  fill: rgba(2,18,21,1);
  width: inherit;
  height: inherit;
}

<WithTheme(Logo)>
  <Logo
    theme={
      Object {
        "borderRadius": "6px",
        "colors": Object {
          "error.100": "#ffd1da",
          "error.200": "#ffa4b6",
          "error.25": "#fff3f5",
          "error.300": "#ff7692",
          "error.400": "#ff496e",
          "error.50": "#ffe8ec",
          "error.500": "#ff1c4a",
          "error.600": "#cc163b",
          "error.700": "#99102c",
          "error.800": "#660b1d",
          "error.900": "#33050e",
          "error.950": "#190207",
          "info.100": "#dae7f7",
          "info.200": "#b5d0ef",
          "info.25": "#f5f9fd",
          "info.300": "#91b9e8",
          "info.400": "#6ca2e0",
          "info.50": "#ecf3fb",
          "info.500": "#488bd9",
          "info.600": "#396fad",
          "info.700": "#2b5382",
          "info.800": "#1c3756",
          "info.900": "#0e1b2b",
          "info.950": "#070d15",
          "primary.100": "#ccf2ed",
          "primary.200": "#99e5db",
          "primary.25": "#f2fbfa",
          "primary.300": "#66d9c9",
          "primary.400": "#32ccb7",
          "primary.50": "#e5f8f6",
          "primary.500": "#00c0a5",
          "primary.600": "#009984",
          "primary.700": "#007363",
          "primary.800": "#004c42",
          "primary.900": "#002621",
          "primary.950": "#001310",
          "secondary.100": "#cedee1",
          "secondary.200": "#9ebdc4",
          "secondary.25": "#f2f6f7",
          "secondary.300": "#6d9ca7",
          "secondary.400": "#3d7a8a",
          "secondary.50": "#e6eef0",
          "secondary.500": "#0d5a6d",
          "secondary.600": "#0a4857",
          "secondary.700": "#073641",
          "secondary.800": "#05242b",
          "secondary.900": "#021215",
          "secondary.950": "#01080a",
          "shade.0": "#ffffff",
          "shade.100": "#e3e3e3",
          "shade.1000": "#000000",
          "shade.200": "#c8c8c8",
          "shade.25": "#f8f8f8",
          "shade.300": "#adadad",
          "shade.400": "#929292",
          "shade.50": "#f1f1f1",
          "shade.500": "#777777",
          "shade.600": "#5f5f5f",
          "shade.700": "#474747",
          "shade.800": "#2f2f2f",
          "shade.900": "#171717",
          "shade.950": "#0b0b0b",
          "success.100": "#dbf2cc",
          "success.200": "#b8e599",
          "success.25": "#f6fbf2",
          "success.300": "#94d966",
          "success.400": "#71cc32",
          "success.50": "#edf8e5",
          "success.500": "#4ec000",
          "success.600": "#3e9900",
          "success.700": "#2e7300",
          "success.800": "#1f4c00",
          "success.900": "#0f2600",
          "success.950": "#071300",
          "tertiary.100": "#ffe2df",
          "tertiary.200": "#ffc5c0",
          "tertiary.25": "#fff7f7",
          "tertiary.300": "#ffa9a1",
          "tertiary.400": "#ff8c82",
          "tertiary.50": "#fff0ef",
          "tertiary.500": "#ff7063",
          "tertiary.600": "#cc594f",
          "tertiary.700": "#99433b",
          "tertiary.800": "#662c27",
          "tertiary.900": "#331613",
          "tertiary.950": "#190b09",
          "warn.100": "#fcddcc",
          "warn.200": "#f9bc99",
          "warn.25": "#fef6f2",
          "warn.300": "#f69a66",
          "warn.400": "#f37932",
          "warn.50": "#fdeee5",
          "warn.500": "#f05800",
          "warn.600": "#c04600",
          "warn.700": "#903400",
          "warn.800": "#602300",
          "warn.900": "#301100",
          "warn.950": "#170800",
        },
        "logo": Object {
          "logoName": "LendiLogo",
        },
        "typography": Object {
          "body": Object {
            "fontFamily": "'Open Sans', sans-serif",
          },
          "heading": Object {
            "fontFamily": "'Cabin', sans-serif",
          },
        },
      }
    }
  >
    <LendiLogo
      color="secondary.900"
      height="100px"
      width="160px"
    >
      <styled.span
        color="secondary.900"
        height="100px"
        width="160px"
      >
        <StyledComponent
          color="secondary.900"
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": false,
                "lastClassName": "c0",
                "rules": Array [
                  "
  width: ",
                  [Function],
                  ";
  height: ",
                  [Function],
                  ";
  vertical-align: middle;
  display: inline-block;

  & > svg {
    fill: ",
                  [Function],
                  ";
    width: inherit;
    height: inherit;
  }
",
                ],
              },
              "displayName": "styled.span",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          height="100px"
          width="160px"
        >
          <span
            className="c0"
            color="secondary.900"
            height="100px"
            width="160px"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 1915.11 609.271"
              width="1em"
            >
              <path
                d="M1830.4 186.955h43.565a41.306 41.306 0 0141.145 41.134v25.421h-84.71v-66.555z"
                fill="#00c0a5"
              />
              <path
                d="M693.642 186.955H737.2a40.708 40.708 0 0129.047 11.5 37.7 37.7 0 0112.1 28.429v376.341h-84.7v-416.27zM902.5 409.608h123.43q-2.442-32.675-16.943-48.4-15.74-16.935-44.772-16.938-26.628 0-43.565 16.938-16.959 16.959-18.15 48.4zm145.213 94.383h60.5q-13.321 50.838-50.824 77.445-37.521 27.849-94.389 27.835-68.978 0-107.694-44.778-38.746-43.553-38.726-124.637 0-77.415 37.513-121 38.719-44.757 107.7-44.777 72.606 0 112.539 43.57 38.71 44.784 38.72 124.637v13.311c0 4.048-.416 6.861-1.207 8.472H902.5q1.2 37.516 18.15 55.663 15.714 19.377 47.192 19.363 20.573 0 35.1-8.472a16.9 16.9 0 006.653-4.233c2-2 4.233-4.238 6.652-6.658q1.2-3.63 9.078-9.678 7.857-6.047 22.389-6.058zM1438.091 603.225h-42.358q-18.147 0-29.647-11.5t-11.5-29.648V397.5q0-29.036-17.544-42.352a63.522 63.522 0 00-39.326-13.311 61.025 61.025 0 00-38.726 13.311q-16.959 13.33-16.937 42.352v205.725h-84.703V414.447q0-82.274 42.959-110.119 42.93-27.823 94.989-30.254a182.223 182.223 0 0198.016 27.834q44.757 27.856 44.777 112.539v188.778zM1557.406 439.856q0 48.422 18.15 73.819 18.156 25.415 52.037 25.415 32.665 0 50.824-25.415 16.926-25.405 16.937-73.819 0-43.554-18.15-68.974-19.368-24.185-52.031-24.2-31.486 0-49.617 25.42-18.147 24.217-18.15 67.754zm137.948-254.119h41.146q16.934 0 29.041 12.716 12.084 12.69 12.1 28.429v235.969q0 85.938-49.617 116.172a203.093 203.093 0 01-107.694 30.248 190.719 190.719 0 01-61.715-10.291q-30.265-10.265-48.4-33.268-35.116-44.749-35.094-122.224 0-76.234 35.094-122.211 36.3-47.2 93.176-47.2 30.231 0 53.243 12.1a115.061 115.061 0 0138.72 37.513V185.737zM1830.4 281.338h43.565a41.317 41.317 0 0141.145 41.146v280.741h-84.71V281.338z"
              />
              <path
                d="M513.617 502.385H363.694V346.877H256.8v155.508H106.886v-250.65l207.028-134.282 199.7 133.772v251.16zm83.189-324.092L344.129 9.041a53.445 53.445 0 00-58.829-.433L24.359 177.86A53.443 53.443 0 000 222.7v333.125a53.443 53.443 0 0053.44 53.446h513.622a53.444 53.444 0 0053.446-53.446V222.7a53.437 53.437 0 00-23.7-44.407z"
                fill="#00c0a5"
              />
            </svg>
          </span>
        </StyledComponent>
      </styled.span>
    </LendiLogo>
  </Logo>
</WithTheme(Logo)>
`;
